test_user.py

```diff
@@ -1,6 +1,21 @@
 from board import Board, BoardState
 from user import User, UserActions 
+from unittest import TestCase
+from unittest.mock import patch
 
+class TestPickGameMode(TestCase):
+    @patch('ui.CommandLinePrompt.get_input', return_value="1")
+    def test_user_selects_first_option(self, input):
+        self.assertEqual(UserActions.pick_game_mode(), "1")
+    
+    @patch('ui.CommandLinePrompt.get_input', return_value="2")
+    def test_user_selects_first_option(self, input):
+        self.assertEqual(UserActions.pick_game_mode(), "2")
+    
+    @patch('ui.CommandLinePrompt.get_input', return_value="3")
+    def test_user_selects_first_option(self, input):
+        self.assertEqual(UserActions.pick_game_mode(), "3")
+    
```

Comment:
This is good, but it makes me wonder why we are even testing it. What we are testing is validation logic, but the name of this function doesn't point to that at all.